src/recon.rs

```diff
@@ -2532,7 +2532,7 @@ pub(crate) unsafe fn rav1d_recon_b_intra<BD: BitDepth>(
                     let index = (((t.by as isize >> 1) + (t.bx as isize & 1)) * (f.b4_stride >> 1)
                         + ((t.bx >> 1) + (t.by & 1)) as isize)
                         as isize;
-                    f.frame_thread.pal[index as usize][0].as_ptr()
+                    f.frame_thread.pal.index(index as usize)[0].as_ptr()
```

Comment:
Is this a valid way to use `DisjointMut`? i.e. without sticking the guard on the stack? It looks like this logic is getting a pointer to one of the nested arrays within `pal`, but you're not keeping the guard active while that pointer is being used. Is this actually going to catch incorrectly overlapping borrows?

Reply:
Fixed